File: packages/oxlint-plugin-react-doctor/src/plugin/rules/react-native/rn-prefer-content-inset-adjustment.ts
import { defineRetiredRule } from "../../utils/define-retired-rule.js";

export const rnPreferContentInsetAdjustment = defineRetiredRule({
  id: "rn-prefer-content-inset-adjustment",
  title: "Manual safe-area insets can duplicate offsets",
  tags: ["test-noise"],
  requires: ["react-native"],
  severity: "warn",
  recommendation:
    "Retired: SafeAreaView wrappers are valid; prefer native content inset adjustment only when manual inset plumbing causes scroll jumps or duplicated safe-area offsets.",
});

File: native/oxlint/rules/rn-prefer-content-inset-adjustment.rs
use oxc_macros::declare_oxc_lint;

use crate::rule::Rule;

#[derive(Debug, Default, Clone)]
pub struct RnPreferContentInsetAdjustment;

declare_oxc_lint!(
    /// Retired React Native content inset adjustment rule.
    RnPreferContentInsetAdjustment,
    react_doctor_native,
    correctness,
    version = "0.1.0",
    short_description = "Retired React Native content inset adjustment rule.",
);

impl Rule for RnPreferContentInsetAdjustment {}
